pkg/proxy/iptables/proxier.go

```diff
@@ -196,7 +196,15 @@ func newServiceInfo(svcPortName proxy.ServicePortName, port *api.ServicePort, se
 	}
 	var stickyMaxAgeSeconds int
 	if service.Spec.SessionAffinity == api.ServiceAffinityClientIP {
-		stickyMaxAgeSeconds = int(*service.Spec.SessionAffinityConfig.ClientIP.TimeoutSeconds)
+		// We changed the API definition that sessionAffinityConfig is required when
```

Comment:
Isn't that set in defaulting?  When we read it back from API it should get default values.  Before we merge this, I need to understand why that isn't working.

Reply:
> Isn't that set in defaulting?  When we read it back from API it should get default values.

When we set the default value in pkg/api/v1/defaults.go, we meet the UT issues.

```
--- FAIL: TestRoundTripTypes (0.31s)

roundtrip.go:326: Service: diff: 
		object.Spec.Selector:
		  a: map[string]string{}
		  b: map[string]string(nil)
		object.Spec.ExternalIPs:
		  a: []string{}
		  b: []string(nil)
		object.Spec.SessionAffinityConfig:
		  a: <nil>
		  b: &api.SessionAffinityConfig{ClientIP:(*api.ClientIPConfig)(0xc420643f48)}
		object.Spec.LoadBalancerSourceRanges:
		  a: []string{}
		  b: []string(nil)
		object.Status.LoadBalancer.Ingress:
		  a: []api.LoadBalancerIngress{}
		  b: []api.LoadBalancerIngress(nil)
```

> It looks like a testcase that goes through defaulting, which causes the diff to fail. Unfortunately, we don't have a great general solution to this (yet) except to patch up the test case to not depend on defaulting. Set that value explicitly.

So, we set the value explicitly which means the timeout value is required when session affinity type is client ip. For example, we set it explicitly in pkg/master/controller.go.

I guess your may refer to
```
// ClientIPConfig represents the configurations of Client IP based session affinity.
type ClientIPConfig struct {
	// timeoutSeconds specifies the seconds of ClientIP type session sticky time.
	// The value must be >0 && <=86400(for 1 day) if ServiceAffinity == "ClientIP".
	// Default value is 10800(for 3 hours).
	// +optional
	TimeoutSeconds *int32
}
```

The comment message maybe not proper.